packages/relay-kit/src/packs/safe-4337/Safe4337Pack.ts

```diff
@@ -220,14 +274,36 @@ export class Safe4337Pack extends RelayKitBasePack<{
       })
     }
 
-    const adjustEstimationData = await feeEstimator?.adjustEstimation?.({
-      bundlerUrl: this.#BUNDLER_URL,
-      entryPoint: this.#ENTRYPOINT_ADDRESS,
-      userOperation: safeOperation.toUserOperation()
-    })
+    const usePaymaster = userOperation.paymasterAndData !== '0x'
+    const hasSafeDeployment = userOperation.initCode !== '0x'
 
-    if (adjustEstimationData) {
-      safeOperation.addEstimations(adjustEstimationData)
+    // adjustment only needed for deployments without paymaster
+    if (!usePaymaster && hasSafeDeployment) {
+      const adjustEstimationData = await feeEstimator?.adjustEstimation?.({
+        bundlerUrl: this.#BUNDLER_URL,
+        entryPoint: this.#ENTRYPOINT_ADDRESS,
+        userOperation: safeOperation.toUserOperation()
+      })
+
+      if (adjustEstimationData) {
+        safeOperation.addEstimations(adjustEstimationData)
+      }
+    }
+
+    if (usePaymaster) {
+      if (!this.#PAYMASTER_URL) {
```

Comment:
Is the paymaster url different than the bundler one? Just checking if the prop is needed or the url is going to remain the same across providers

Reply:
Yes, it seems that now in the docs, they are using the same v2 endpoint for both clients:

```
https://docs.pimlico.io/permissionless/tutorial/tutorial-1#create-the-bundler-and-smart-account-clients-and-fetch-the-gas-price
```